### DISCUSSION TITLE
How to use the new way to customize component style

### DISCUSSION THREAD

**Discussion by IronTony** [chakra-ui/chakra-ui#4730] [Q&A]
### Subject

Tabs/Input/Menu custom style

### Description

Hi all,

I have some problem understand how to use the new method to  add custom style to components.
I have read the documentation here, and for the Button component is quite easy to follow, but for Input, (I have done but I lost a lot of time to understand),  Tabs, Menu and so on I am really lost.

I think would be great to enrich the documentation with more examples.

All the best

**Comment by segunadebayo** [comment]
@lazarnikolov94 Can you help @IronTony better understand this?

I'll move this to a discussion in the meantime.

**Comment by nikolovlazar** [comment]
Hey @IronTony! The Tabs and Menu are Multi-Part Components. That means when creating a custom style for them you need to specify which part you're overriding. The Menu component for example has the following parts: `button`, `list`, `item`, `groupTitle`, `command`, `divider`.

You can style each part individually by wrapping their style configs in their part name. Here's how you can override the [Menu component](https://chakra-ui.com/docs/theming/component-style#styling-multipart-components). And here's a [working CodeSandbox](https://codesandbox.io/s/menu-item-text-color-mode-k70sj?file=/src/index.tsx) to play with it.

Currently, we don't have the component parts displayed in the docsite, but you can check them out in the `anatomy` package here: https://github.com/chakra-ui/chakra-ui/blob/main/packages/anatomy/src/index.ts.

**Reply by IronTony** [reply]
Hi @lazarnikolov94 and thank you so much for the example.
I finally styled correctly my custom Tab, and the problem was the name of the component in the "customTheme" variable.
I wrote this:
``` 
components: {
    Button,
    Input,
    Tab,
  },
```

instead of:
```
components: {
    Button,
    Input,
    Tabs,
  },
```
and that's why I could not use the style.

I just have a small problem always on this:

I am trying to change the style also of the selected action, in this way:

```
baseStyle: {
    tab: {
      borderTopRadius: '10px',
      color: 'grey2',
      fontWeight: 400,
      height: '60px',
      _selected: {
        bg: 'grey4',
        borderBottomWidth: '3px',
        borderBottomColor: 'primary', <<<<<<<< not working
        color: 'black', <<<<<<<< not working
      },
    },
  },
```

But the selected is not working, any hint about that?

Thank you so much

**Comment by IronTony** [comment]
Moreover, is there a way to change some colors and so on of the Toast?
I think that is not possible to use the above flow also for the toast. Am I wrong?

**Comment by maxskewes** [comment]
The "working sandbox" isn't working unfortunately, however the code doesn't provide a lot of insight as to the potential styling of the Menu multi-component. I just want to know what props I can put in and where to put them to style the most common pieces; alas there's no list on the Menu page. I really wish a part of each page on the Chakra site took some time to address styling or at least varied the examples to include a wider range of options. For example, I think a standard modification would be for the menu items to justify to the right. So logically, the MenuList component would contain a prop for this. If it does, I cannot guess it. I would also like to know how to change the hover color. Thanks if anyone can clear any of this up for me or point me in the direction of a functional component (or group) that Chakra's Menu may have been based on that may have more thorough documentation.

**Comment by lautiab** [comment]
@nikolovlazar Sorry to re-bump this but I'm struggling with a pretty similar thing. I'd like to take advantage of Menu component customization the way you've described, but not affecting every Menu used in the app. 

Thought of using sth like `chakra(Menu, {baseStyle: {button: {bg: 'red', item: {color: 'red'}}}})` but is not working.. I infer chakra factory might be trying to add styles to the Menu wrapper or sth similar

Is there a way to accomplish this without re-using inline styles each time I want to use the customized component?
